File: src/main/java/com/gmail/borlandlp/minigamesdtools/activepoints/behaviors/custom/ExampleBehaviorCreator.java
package com.gmail.borlandlp.minigamesdtools.activepoints.behaviors.custom;

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools;
import com.gmail.borlandlp.minigamesdtools.activepoints.ActivePoint;
import com.gmail.borlandlp.minigamesdtools.config.ConfigPath;
import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider;
import com.gmail.borlandlp.minigamesdtools.creator.Creator;
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo;
import org.bukkit.configuration.ConfigurationSection;

import java.util.ArrayList;
import java.util.List;

@CreatorInfo(creatorId = "example_behavior")
public class ExampleBehaviorCreator extends Creator {
    @Override
    public Object create(String Id, AbstractDataProvider dataProvider) throws Exception {
        ExampleBehavior exampleBehavior = new ExampleBehavior();
        ConfigurationSection conf = MinigamesDTools.Companion.getInstance().getConfigProvider().getEntity(ConfigPath.ACTIVE_POINT_BEHAVIORS, Id).getData();
        if(conf == null) {
            throw new Exception("Cant find config for Behavior[ID:" + Id + "]");
        }

        exampleBehavior.setActivePoint((ActivePoint) dataProvider.get("active_point_instance"));
        exampleBehavior.setName(Id);
        exampleBehavior.setDelay(Integer.parseInt(conf.get("delay").toString()));
        exampleBehavior.setPeriod(Integer.parseInt(conf.get("period").toString()));

        return exampleBehavior;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/activepoints/behaviors/custom/ExampleBehaviorCreator.kt
package com.gmail.borlandlp.minigamesdtools.activepoints.behaviors.custom

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools.Companion.instance
import com.gmail.borlandlp.minigamesdtools.activepoints.ActivePoint
import com.gmail.borlandlp.minigamesdtools.config.ConfigPath
import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider
import com.gmail.borlandlp.minigamesdtools.creator.Creator
import com.gmail.borlandlp.minigamesdtools.creator.CreatorInfo

@CreatorInfo(creatorId = "example_behavior")
class ExampleBehaviorCreator : Creator() {
    @Throws(Exception::class)
    override fun create(id: String, dataProvider: AbstractDataProvider): Any {

        val conf =
            instance!!.configProvider!!.getEntity(ConfigPath.ACTIVE_POINT_BEHAVIORS, id)?.data
                ?: throw Exception("Cant find config for Behavior[ID:$id]")
        val activePoint = (dataProvider["active_point_instance"] as ActivePoint)
        val delay = conf["delay"].toString().toInt()
        val period = conf["period"].toString().toInt()
        return ExampleBehavior(id, activePoint, delay, period)
    }
}